includes/class-amp-reader-themes.php

```diff
@@ -0,0 +1,546 @@
+	 * @since 1.6.0
+	 *
+	 * @var string
+	 */
+	const INSTALLED_STATUS = 'installed';
+
+	/**
+	 * Status indicating a reader theme is not installed but is installable.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @var string
+	 */
+	const INSTALLABLE_STATUS = 'installable';
+
+	/**
+	 * Status indicating a reader theme is not installed and can't be installed.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @var string
+	 */
+	const NON_INSTALLABLE_STATUS = 'non-installable';
+
+	/**
+	 * Retrieves all AMP plugin options specified in the endpoint schema.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @return array Formatted theme data.
+	 */
+	public function get_themes() {
+		if ( ! is_null( $this->themes ) ) {
+			return $this->themes;
+		}
+
+		$themes   = $this->get_default_supported_reader_themes();
+		$themes   = array_map( [ $this, 'prepare_theme' ], $themes );
+		$themes[] = $this->get_classic_mode();
+
+		/**
+		 * Filters supported reader themes.
+		 *
+		 * @param array $themes [
+		 *     Reader theme data.
+		 *     {
+		 *         @type string         $name           Theme name.
+		 *         @type string         $slug           Theme slug.
+		 *         @type string         $slug           URL of theme preview.
+		 *         @type string         $screenshot_url The URL of a mobile screenshot. Note: if this is empty, the theme may not display.
+		 *         @type string         $homepage        A link to a page with more information about the theme.
+		 *         @type string         $description     A description of the theme.
+		 *         @type string|boolean $requires        Minimum version of WordPress required by the theme. False if all versions are supported.
+		 *         @type string|boolean $requires_php    Minimum version of PHP required by the theme. False if all versions are supported.
+		 *         @type string         $download_link   A link to the theme's zip file. If empty, the plugin will attempt to download the theme from wordpress.org.
+		 *     }
+		 * ]
+		 */
+		$themes = apply_filters( 'amp_reader_themes', $themes );
+
+		foreach ( $themes as &$theme ) {
+			$theme['availability'] = $this->get_theme_availability( $theme );
+		}
+
+		$this->themes = $themes;
+
+		return $this->themes;
+	}
+
+	/**
+	 * Gets a reader theme by slug.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param string $slug Theme slug.
+	 * @return array Theme data.
+	 */
+	public function get_reader_theme( $slug ) {
+		return current(
+			array_filter(
+				$this->get_themes(),
+				static function( $theme ) use ( $slug ) {
+					return $theme['slug'] === $slug;
+				}
+			)
+		);
+	}
+
+	/**
+	 * Retrieves theme data.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param boolean $from_api Whether to return theme data from the wordpress.org API. Default false.
+	 * @return array Theme ecosystem posts copied the amp-wp.org website.
+	 */
+	public function get_default_supported_reader_themes( $from_api = false ) {
+		// Note: This can be used to refresh the hardcoded raw theme data.
+		if ( $from_api ) {
```

Comment:
The logic should be inverted to avoid indented almost the entire method body.
```suggestion
		if ( ! $from_api ) {
			return self::get_default_raw_reader_themes();
		}
		
		// API calls here...
```